### DISCUSSION TITLE
Event Requeuing (EventReader/EventWriter)

### DISCUSSION THREAD

**Discussion by Wolftousen** [bevyengine/bevy#2442] [Q&A] [answered]
I'm curious how to go about requeuing events with EventReader/EventWriter.

For example, I made snake as a "learn bevy" project and got everything working.  I then decided I wanted to add fruits that do different things and my first one was, grow 3 instead of 1.  The way I originally planned to do this was by queuing a Grow(3) event where struct Grow (i32) is the event type.  Then when I read that event, I grow and queue a new event Grow(2) and so on until I have no grow amount left.

I ran into the issue that you cannot send mut EventReader<T> and mut EventWriter<T> to a system and will cause a runtime panic.

I'm curious what the ideal/expected solution for this would be if anyone knows.

**Comment by alice-i-cecile** [comment, marked as answer]
You can work with `ManualEventReader<T>` (I think) or the raw `ResMut<Events<T>>` to do this. This isn't the first time this has come up; I think there should be an example demonstrating this functionality.

**Comment by ManevilleF** [comment]
I ran into the same issue with my learning project as well, I suggest using components for delayed elements instead of events.
